pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -199,31 +207,65 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 			v = workerPoolSemver.String()
 			maintainedShoot.Spec.Provider.Workers[i].Kubernetes.Version = &v
 			return nil
-		}, name)
+		})
 		if err != nil {
-			// continue execution to allow the machine image version update
-			workerLog.Error(err, "Could not maintain kubernetes version for worker pool")
+			// continue execution to allow other maintenance activities to continue
+			workerLog.Error(err, "Could not maintain Kubernetes version for worker pool")
+		}
+
+		if reasonForUpdate != nil {
+			workerToReasonForKubernetesUpdate[pool.Name] = *reasonForUpdate
+		}
+
+		if successMsg != nil {
+			workerToSuccessfulKubernetesUpdate[pool.Name] = *successMsg
 		}
 
-		reasonForKubernetesUpdate = append(reasonForKubernetesUpdate, reasonForWorkerPoolKubernetesUpdate...)
+		if failureMsg != nil {
+			workerToFailedKubernetesUpdate[pool.Name] = *failureMsg
+		}
 	}
-	operations = append(operations, reasonForKubernetesUpdate...)
 
 	operation := maintainOperation(maintainedShoot)
 	if operation != "" {
 		operations = append(operations, fmt.Sprintf("Added %q operation annotation", operation))
 	}
 
-	if len(operations) > 0 {
+	requirePatch := len(operations) > 0 || reasonControlPlaneKubernetesUpdate != nil || len(workerToReasonForKubernetesUpdate) > 0 || len(workerToReasonForImageUpdate) > 0
+	if requirePatch {
 		patch := client.MergeFrom(shoot.DeepCopy())
+
+		// make sure to both include successful and failed maintenance operations
+		description, failureReason := buildMaintenanceMessages(
+			reasonControlPlaneKubernetesUpdate,
+			successMsgControlPlaneKubernetesUpdate,
+			failureMsgControlPlaneKubernetesUpdate,
+			workerToReasonForKubernetesUpdate,
+			workerToSuccessfulKubernetesUpdate,
+			workerToFailedKubernetesUpdate,
+			workerToReasonForImageUpdate,
+			workerToSuccessfulImageUpdate,
+			workerToFailedImageUpdate)
+
+		// append also other maintenance operation
+		if len(operations) > 0 {
+			description = fmt.Sprintf("%s, %s", description, strings.Join(operations, ", "))
+		}
+
 		shoot.Status.LastMaintenance = &gardencorev1beta1.LastMaintenance{
-			Description:   strings.Join(operations, ", "),
+			Description:   description,
 			TriggeredTime: metav1.Time{Time: r.Clock.Now()},
-			State:         gardencorev1beta1.LastOperationStateProcessing,
+			State:         gardencorev1beta1.LastOperationStateSucceeded,
```

Comment:
We can set the state here to succeed, what if at a later stage, maintenance gets failed? At this stage, it can't be made sure that changes were applied to shoot.
```suggestion
			State:         gardencorev1beta1. LastOperationStateProcessing,
```